scripts_regression_tests cleanup is not working

In this case a test fails but it removes the directory anyway:
```
======================================================================
FAIL: test_clean_rebuild (__main__.G_TestBuildSystem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./scripts_regression_tests.py", line 2558, in test_clean_rebuild
    casedir = self._create_test(["--no-run", "SMS.f19_g16_rx1.A"], test_id=self._baseline_name)
  File "./scripts_regression_tests.py", line 1267, in _create_test
    expected_stat=expected_stat)
  File "./scripts_regression_tests.py", line 71, in run_cmd_assert_result
    test_obj.assertEqual(stat, expected_stat, msg=msg)
AssertionError: 100 != 0 : 
COMMAND:  /home/jedwards/cesm/scratch/cime-nightly-build/cime/scripts/create_test --no-run SMS.f19_g16_rx1.A -t fake_testing_only_20201217_100033 --baseline-root /home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033/baselines --test-root=/home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033 --output-root=/home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033
FROM_DIR: /home/jedwards/cesm/scratch/cime-nightly-build/cime/scripts/tests
SHOULD HAVE WORKED, INSTEAD GOT STAT 100
OUTPUT: Testnames: ['SMS.f19_g16_rx1.A.centos7-linux_gnu']
No project info available
create_test will do up to 1 tasks simultaneously
create_test will use up to 10 cores simultaneously
Creating test directory /home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033/SMS.f19_g16_rx1.A.centos7-linux_gnu.fake_testing_only_20201217_100033
RUNNING TESTS:
  SMS.f19_g16_rx1.A.centos7-linux_gnu
Starting CREATE_NEWCASE for test SMS.f19_g16_rx1.A.centos7-linux_gnu with 1 procs
Finished CREATE_NEWCASE for test SMS.f19_g16_rx1.A.centos7-linux_gnu in 0.888158 seconds (PASS)
Starting XML for test SMS.f19_g16_rx1.A.centos7-linux_gnu with 1 procs
Finished XML for test SMS.f19_g16_rx1.A.centos7-linux_gnu in 0.250136 seconds (PASS)
Starting SETUP for test SMS.f19_g16_rx1.A.centos7-linux_gnu with 1 procs
Finished SETUP for test SMS.f19_g16_rx1.A.centos7-linux_gnu in 0.876012 seconds (PASS)
Starting SHAREDLIB_BUILD for test SMS.f19_g16_rx1.A.centos7-linux_gnu with 1 procs
Finished SHAREDLIB_BUILD for test SMS.f19_g16_rx1.A.centos7-linux_gnu in 38.482795 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033/SMS.f19_g16_rx1.A.centos7-linux_gnu.fake_testing_only_20201217_100033
    Errors were:
        Building test for SMS in directory /home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033/SMS.f19_g16_rx1.A.centos7-linux_gnu.fake_testing_only_20201217_100033
           NOTE: ignoring setting of rof2ocn_fmapname=idmap in seq_maps.rc
        ERROR: /home/jedwards/cesm/scratch/cime-nightly-build/cime/src/build_scripts/buildlib.csm_share FAILED, cat /home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033/SMS.f19_g16_rx1.A.centos7-linux_gnu.fake_testing_only_20201217_100033/bld/csm_share.bldlog.201217-100036

At test-scheduler close, state is:
FAIL SMS.f19_g16_rx1.A.centos7-linux_gnu (phase SHAREDLIB_BUILD)
    Case dir: /home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033/SMS.f19_g16_rx1.A.centos7-linux_gnu.fake_testing_only_20201217_100033
test-scheduler took 41.06489849090576 seconds
ERRPUT: 


----------------------------------------------------------------------
Ran 1 test in 41.323s

FAILED (failures=1)
All pass, removing directory: /home/jedwards/cesm/scratch/scripts_regression_test.20201217_100033
```

--- a/scripts/tests/scripts_regression_tests.py
+++ b/scripts/tests/scripts_regression_tests.py
@@ -47,6 +47,7 @@ NO_CMAKE    = False
 TEST_ROOT   = None
 NO_TEARDOWN = False
 NO_FORTRAN_RUN      = False
+TEST_RESULT = None
 
 os.environ["CIME_GLOBAL_WALLTIME"] = "0:05:00"
 
@@ -3511,7 +3512,7 @@ def write_provenance_info():
 
 def cleanup():
     # if the TEST_ROOT directory exists and is empty, remove it
-    if os.path.exists(TEST_ROOT):
+    if os.path.exists(TEST_ROOT) and TEST_RESULT.result.wasSuccessful():
         testreporter = os.path.join(TEST_ROOT,"testreporter")
         files = os.listdir(TEST_ROOT)
         if len(files)==1 and os.path.isfile(testreporter):
@@ -3531,6 +3532,7 @@ def _main_func(description):
     global GLOBAL_TIMEOUT
     global NO_TEARDOWN
     global NO_FORTRAN_RUN
+    global TEST_RESULT
     config = CIME.utils.get_cime_config()
 
     help_str = \
@@ -3662,12 +3664,15 @@ OR
             setattr(B_CheckCode, testname, pylint_test)
 
     try:
-        unittest.main(verbosity=2, catchbreak=True)
+        TEST_RESULT = unittest.main(verbosity=2, catchbreak=True, exit=False)
     except CIMEError as e:
         if e.__str__() != "False":
             print("Detected failures, leaving directory:", TEST_ROOT)
             raise
-
+    else:
+        # Implements same behavior as unittesst.main
+        # https://github.com/python/cpython/blob/b6d68aa08baebb753534a26d537ac3c0d2c21c79/Lib/unittest/main.py#L272-L273
+        sys.exit(not TEST_RESULT.result.wasSuccessful())
 
 if (__name__ == "__main__"):
     _main_func(__doc__)

